store/store.go

```diff
@@ -1093,6 +1100,48 @@ func (s *Store) Find(search *Search, user *auth.UserState) ([]*snap.Info, error)
 	return snaps, nil
 }
 
+// SnapCommands retrieves a mapping of snap names to the commands it provides.
+func (s *Store) SnapCommands() (map[string][]string, error) {
+	u := *s.commandsURI
+
+	q := u.Query()
+	if release.OnClassic {
+		q.Set("confinement", "strict,classic")
+	} else {
+		q.Set("confinement", "strict")
+	}
+
+	u.RawQuery = q.Encode()
+	reqOptions := &requestOptions{
+		Method: "GET",
+		URL:    &u,
+		Accept: halJsonContentType,
+	}
+
+	var cmdsData cmdsResults
+	resp, err := s.retryRequestDecodeJSON(context.TODO(), reqOptions, nil, &cmdsData, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != 200 {
+		return nil, respToError(resp, "commands")
+	}
+
+	if ct := resp.Header.Get("Content-Type"); ct != halJsonContentType {
+		return nil, fmt.Errorf("received an unexpected content type (%q) when trying to retrieve the commands via %q", ct, resp.Request.URL)
+	}
+
+	cmds := make(map[string][]string, len(cmdsData.Payload.Packages))
```

Comment:
do we need a TODO here about the fact that at moment we get only snap names?

Reply:
I'll add it (although we also get aliases right now)